app/src/main/java/org/groundplatform/android/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -281,21 +283,47 @@ class GoogleMapsFragment : SupportMapFragment(), MapFragment {
 
   override fun addTileOverlay(source: TileSource) =
     when (source) {
-      is LocalTileSource -> addLocalTileOverlay(source.localFilePath, source.clipBounds)
-      is RemoteMogTileSource -> addRemoteMogTileOverlay(source.remotePath)
+      is LocalTileSource ->
+        addLocalTileOverlay(source.localFilePath, source.clipBounds, source.maxZoom)
+      is RemoteMogTileSource ->
+        addRemoteMogTileOverlay(url = source.remotePath, zIndex = TILE_OVERLAY_Z)
     }
 
-  private fun addLocalTileOverlay(url: String, bounds: List<Bounds>) {
-    addTileOverlay(
-      ClippingTileProvider(TemplateUrlTileProvider(url), bounds.map { it.toGoogleMapsObject() })
-    )
+  private fun addLocalTileOverlay(url: String, bounds: List<Bounds>, maxZoom: Int) {
+    val baseProvider = TemplateUrlTileProvider(url)
+    // Upscale from the max downloaded zoom level to higher levels (e.g., 15, 16, 17, 18) by
+    // splitting each tile into four quadrants and doubling each pixel to create 256x256 output
+    // tiles
+    val upscaledProvider = CachingUpscalingTileProvider(baseProvider, dataMaxZoom = maxZoom)
+    val clippedProvider =
+      ClippingTileProvider(upscaledProvider, bounds.map { it.toGoogleMapsObject() })
+    addTileOverlay(clippedProvider)
   }
 
-  private fun addRemoteMogTileOverlay(url: String) {
-    // TODO: Make sub-paths configurable and stop hardcoding here.
-    // Issue URL: https://github.com/google/ground-android/issues/1730
+  private fun addRemoteMogTileOverlay(
+    url: String,
+    dataMaxZoom: Int = 15,
```

Comment:
Can we remove the defaults from here and consolidate all defaults to one location? If this is some other default value, then lets move it to companion object and rename it appropriately.